detox/src/devices/drivers/android/attached/FreeAdbDeviceFinder.test.js

```diff
@@ -0,0 +1,17 @@
+const FreeAdbDeviceFinder = require('./FreeAdbDeviceFinder');
+
+describe('FreeAdbDeviceFinder', () => {
+  it("should test candidate's adbName against a regular expression", async () => {
+    const mockRegExpTest = jest.fn();
```

Comment:
Maybe I am missing something in this life, but can't we just duck test it, something like this?

```js
const onlyDigits =  "^\\d+$";

expect(await finder.isDeviceMatching({ adbName: '5555' }, onlyDigits).toBe(true);
expect(await finder.isDeviceMatching({ adbName: 'emulator-5555' }, onlyDigits).toBe(false);
```

Reply:
Changed